pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Producer.java

```diff
@@ -138,6 +138,15 @@ public Producer(Topic topic, TransportCnx cnx, long producerId, String producerN
         this.clientAddress = cnx.clientSourceAddress();
     }
 
+    private String parseRemoteClusterName(String producerName, boolean isRemote) {
+        if (isRemote) {
+            String clusterName = producerName.split("\\.")[2];
+            return clusterName.contains(REPL_PRODUCER_NAME_DELIMITER)
+                    ? clusterName.split(REPL_PRODUCER_NAME_DELIMITER)[1] : clusterName;
```

Comment:
Actually, I think previous ` clusterName.split(REPL_PRODUCER_NAME_DELIMITER)[0]` should be right.
For replicator A-->B, this producer is in cluster B, it's remoteCluster is A.

Reply:
Yes， right. Thanks.